packages/connect/src/SmoldotProvider/SmoldotProvider.ts

```diff
@@ -280,34 +281,33 @@ export class SmoldotProvider implements ProviderInterface {
           }
         }
       })
-      .then((client: smoldot.SmoldotClient) => {
-        this.#client = client;
-        this.#connectionStatePingerId = setInterval(
-          this.#checkClientPeercount, this.healthPingerInterval);
-      })
-      .catch((error: Error) => {
-        this.emit('error', error);
-        return Promise.reject(error);
-      });
+      this.#connectionStatePingerId = setInterval(
+      this.#checkClientPeercount, this.healthPingerInterval);
+    } catch(error: unknown) {
+      this.emit('error', error);
+    }
   }
 
   /**
    * @description Manually "disconnect" - drops the reference to the WASM client
    */
   // eslint-disable-next-line @typescript-eslint/require-await
   public async disconnect(): Promise<void> {
-    if (this.#client) {
-      this.#client = undefined;
-    }
-
-    if (this.#connectionStatePingerId !== null) {
-      clearInterval(this.#connectionStatePingerId);
-    }
+    try {
+      if (this.#client) {
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        this.#client.terminate;
```

Comment:
This should be a function call `this.#client.terminate()`

Reply:
Tried that and for some reason it wont work (Will get error .terminate() is not a function).
You can give it a try - maybe I am missing something

Reply:
This is a new bug.  All the tests  pass calling `terminate()`  except  one which fails with this:

```
Expected: "Got error response asking for system health"
Received: "__classPrivateFieldGet(...).terminate is not a function"".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```
An unhandled rejection error has been assigned to `this.#client` .. so something has gone wrong in the refactoring from Promise control flow to `async` / `await` with `try` / `catch` in the `connect()` function

Reply:
It's also possible this is a new bug in `smoldot.start`  try reverting to using a `Promise` with `then` and `catch` and see what happens if we still end up with an error being assigned to `this.#client` then the problem is in smoldot

Reply:
We should see that on scope of #185 and not on this  ticket.  Lets keep the PR on single scope

Reply:
I agree that we should try to get this merged asap and move onto fixing the outstanding issues to do with jest thinking there is a leak separately.  This thing however is an unrelated and new bug that has been introduced (probably by the refactoring to async/await in connect) and is definitely not the right logic (1. terminate is a function and 2, we weren't assigning errors to `this/#client` before). This means we are neither calling `terminate` (we just have a reference to the property) and we are guaranteed to throw errors to users in some cases.

I recommend we reinstate the old body of `connect()` in `SmoldotProvider.ts` and change `this.#client.terminate;` to `this.#client.terminate();` in `disconnect()` and then move on with merging .. that way disconnect in master isn't broken while we investigate the jest memory leaking issues